src/Data/List.agda

```diff
@@ -11,7 +11,35 @@
 
 module Data.List where
 
+import Data.List.NonEmpty.Base as List⁺
+import Data.List.Properties as List
+open import Function.Base using (_∘_)
+open import Relation.Binary.PropositionalEquality using (refl; cong₂; _≗_)
+
 ------------------------------------------------------------------------
 -- Types and basic operations
 
-open import Data.List.Base public
+open import Data.List.Base public hiding (scanr)
+
+------------------------------------------------------------------------
+-- scanr
+
+module _ {a b} {A : Set a} {B : Set b} (f : A → B → B) (e : B) where
+
+  open List⁺ using (List⁺; _∷_; toList)
+
+-- definition
+
+  scanr : List A → List B
+  scanr = toList ∘ go where
+    go : List A → List⁺ B
+    go []       = e ∷ []
+    go (x ∷ xs) = let y ∷ ys = go xs in f x y ∷ y ∷ ys
+
+-- property
+
+  scanr-defn : scanr ≗ map (foldr f e) ∘ tails
```

Comment:
We shouldn't have properties in definition files, especially for lists.

Reply:
So this is a very irritating `dependencies` problem: 
* I took @gallais 's suggestion to lift this out of `Data.List.NonEmpty.Base` in favour of `Data.List`;
* but the 'property' then gets orphaned, and yet depends on both `NonEmpty` and `List`;
* ... and `Data.List.Properties` is both already taken, and prior to  `Data.List.NonEmpty.Properties`, and the property at hand...

I could revert various things, but not really sure how best to proceed... partly because `scanr` is already an irritating wart, but I don't want to grant it a module on its own for the sake of breaking the dependency cycle.

Reply:
Have attempted a principled refactoring, but with the above redundancy in commentary/documentation.

Reply:
I very much like the refactoring.